plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTask.java

```diff
@@ -168,11 +173,23 @@ public void performAction(IncrementalTaskInputs inputs) throws Exception {
 				.steps(steps)
 				.exceptionPolicy(exceptionPolicy)
 				.build()) {
+			// determine if a list of files has been passed in
+			Predicate<File> shouldInclude;
+			Project project = getProject();
+			if (project.hasProperty(FILES_PROPERTY) && project.property(FILES_PROPERTY) instanceof String) {
+				Object rawIncludePatterns = project.property(FILES_PROPERTY);
+				assert rawIncludePatterns != null;
+				final String[] includePatterns = ((String) rawIncludePatterns).split(",");
+				shouldInclude = file -> Arrays.stream(includePatterns)
+						.anyMatch(filePattern -> file.getAbsolutePath().matches(filePattern));
```

Comment:
This means that the patterns will be compiled for each file checked.

Reply:
updated to compile the patterns once instead of each time the lambda is called